toolkit/offchain/src/d_param.rs

```diff
@@ -0,0 +1,201 @@
+#![allow(dead_code)]
+
+use crate::csl::{
+	add_collateral_inputs, add_mint_script_token, add_output_with_one_script_token,
+	convert_cost_models, get_builder_config, key_hash_address, ogmios_utxos_to_csl,
+};
+use crate::plutus_script::PlutusScript;
+use cardano_serialization_lib::{
+	ChangeConfig, CoinSelectionStrategyCIP2, Ed25519KeyHash, ExUnits, JsError, NetworkIdKind,
+	PlutusData, Transaction, TransactionBuilder,
+};
+use ogmios_client::{query_ledger_state::ProtocolParametersResponse, types::OgmiosUtxo};
+use partner_chains_plutus_data::d_param::DParamDatum;
+use sidechain_domain::DParameter;
+
+#[allow(clippy::too_many_arguments)]
+fn mint_token_tx(
+	validator: &PlutusScript,
+	d_parameter: &DParameter,
+	payment_key_hash: &Ed25519KeyHash,
+	collaterals: &[OgmiosUtxo],
+	payment_utxos: &[OgmiosUtxo],
+	network: NetworkIdKind,
+	protocol_parameters: &ProtocolParametersResponse,
+	mint_witness_ex_units: ExUnits,
```

Comment:
I'm wondering about adding a struct in `csl` module
```
pub(crate) struct ScriptTransactionContext {
  /// script with all parameters applied
  pub(crate)  script: PlutusScript,
  pub(crate) network: NetworkIdKind,
  pub(crate) protocol_parameters: ProtocolParametersResponse,
  }
  ```
 Good idea, more or less? Should I open a separate PR to discuss exact fields of such a field?